src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -177,6 +177,108 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
         }
     };
 
+    template<typename T>
+    struct ConversionTrait<std::unordered_map<std::string, T>>
+    {
+        std::unordered_map<std::string, T> FromJson(const Json::Value& json) const
+        {
+            std::unordered_map<std::string, T> val;
+
+            for (const auto& element : json.getMemberNames())
+            {
+                GetValueForKey(json, element, val[element.c_str()]);
+            }
+
+            return val;
+        }
+
+        bool CanConvert(const Json::Value& json) const
+        {
+            if (!json.isObject())
+            {
+                return false;
+            }
+            ConversionTrait<T> trait;
+            for (const auto& element : json.getMemberNames())
+            {
+                if (!trait.CanConvert(json[JsonKey(element)]))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        Json::Value ToJson(const std::unordered_map<std::string, T>& val)
+        {
+            ConversionTrait<T> trait;
+            Json::Value json{ Json::objectValue };
+
+            for (const auto& [k, v] : val)
+            {
+                json[JsonKey(k)] = trait.ToJson(v);
+            }
+
+            return json;
+        }
+
+        std::string TypeDescription() const
+        {
+            return "map";
+        }
+    };
+
+    template<typename T>
+    struct ConversionTrait<std::unordered_map<std::wstring, T>>
+    {
+        std::unordered_map<std::wstring, T> FromJson(const Json::Value& json) const
+        {
+            std::unordered_map<std::wstring, T> val;
+
+            for (const auto& element : json.getMemberNames())
+            {
+                GetValueForKey(json, element, val[til::u8u16(std::string_view{ element.c_str() })]);

```

Comment:
The compiler is unhappy with `til::u8u16(element.c_str())` (outputs a 'no matching overloaded function found' error), hence the strange conversion to `string_view` from `string` first. If someone knows of a better way to do this that'll be great

Reply:
what is the type of `element`? can you not just use `til::u8u16(element)`?

If you find yourself using c_str when you're moving between different well-known smart string types, generally best to reconsider. :smile:

Reply:
Okay idk why I thought I already tried that >< Thanks, fixed!